docs/DeveloperGuide.md

```diff
@@ -481,6 +481,201 @@ Below is an Activity Diagram illustrating the flow of activities when the applic
 
                                ------------------------------Activity diagram illustrating multithreading (will add later)--------------------------------
 
+### \[Implemented\] Adding an assignment feature
+
+The user can add an assignment by providing keywords of the following fields:
+- Name of assignment
+- Module code of assignment
+- Deadline of assignment (contains both due time and due date)
+- Priority level (Optional)
+- Remind (Optional)
+
+It implements the following operations:
+- `add n/Lab 3 mod/CS2100 d/24-10-2020 2359` - Adds an assignment with name Lab 3, module code CS2100, due time 2359, and with due date 24-10-2020.
+- `add n/Lab 3 mod/CS2100 d/24-10-2020 2359 p/HIGH` - Adds a high priority assignment with name Lab 3, module code CS2100, due time 2359 and with due date 24-10-2020.  
+- `add n/Lab 3 mod/CS2100 d/24-10-2020 2359 p/MEDIUM remind` - Adds a medium priority assignment with name Lab 3, module code CS2100, due time 2359, due date 24-10-2020, and with reminders set.
+
+#### Reasons for Implementation
+
+Being able to add an assignment is essential for the user to manage his/her academic tasks. 
+As a student user, the following scenarios are likely:
+- The user wants to add an assignment with a certain priority level attach to it, so he/she knows how urgent the assignment is.
+- The user wants to add an assignment and wants the assignment to appear directly in his/her Reminders list right away since it's something important.
+- The user wants to add an assignment with a certain priority level and also wants the assignment to appear directly in his/her Reminders list right away.
+
+We decided to allow the user to include the priority level and remind tag if he/she wants to do so. We understand that not all users will 
+know the priority level of the assignment at the point he/she is adding the assignment and thus we made the priority level and remind keywords optional.
+
+#### Current Implementation
+
+##### Prefixes used in identifying keywords
+The use of prefixes before keywords facilitates the extraction of keywords from the user's input and allow keywords to be validated subsequently.
+
+The following prefixes are used to identify the type of keywords:
+- `/n` for Name keyword
+- `/mod` for Module code keyword
+- `/d` for Deadline keyword
+- `/p` for Priority keyword
+- `/remind` for Remind keyword 
```

Comment:
I don't think remind has a prefix in the add command, though I could just be remembering wrongly

Reply:
oh yea remind no prefix, thanks for pointing this out!